site/app/controllers/admin/UsersController.php

```diff
@@ -273,7 +273,7 @@ public function updateUser($type='users') {
             }
         }
         else {
-            if ($user !== null) {
+            if ($user->getRegistrationSection() !== null) {
```

Comment:
Is this the right check?  The registration section can be NULL when a student drops.
(And I think the value is NULL rather than a string NULL)

Reply:
I've tested this and the GUI works correctly in all cases I tested.
So I'm going to merge this since it's an urgent bug.
@zjxiaohan and @MasterOdin please follow up if there are remaining concerns.

Reply:
> Is this the right check? The registration section can be NULL when a student drops.
(And I think the value is NULL rather than a string NULL)

No. I removed in the second commit.
There are remaining concerns though. I will open an issue later.